components/script/dom/node.rs

```diff
@@ -1298,36 +1349,39 @@ impl TreeIterator {
         self.next_skipping_children_impl(current)
     }
 
-    fn next_skipping_children_impl(&mut self, current: Root<Node>) -> Option<Root<Node>> {
+    fn next_skipping_children_impl(&mut self, current: Root<Node>) -> Option<Root<T>> {
         for ancestor in current.inclusive_ancestors() {
             if self.depth == 0 {
                 break;
             }
             if let Some(next_sibling) = ancestor.GetNextSibling() {
-                self.current = Some(next_sibling);
-                return Some(current);
+                self.current = Some(next_sibling.clone());
+                return Root::downcast::<T>(current);
```

Comment:
Same remark about `None`, the iterator should never return `None` until it's done with the whole tree.